src/common/marlin_server.hpp

```diff
@@ -1,16 +1,182 @@
-// marlin_server.hpp
+/**
+ * @file marlin_server.hpp
+ * @brief interface between Marlin and Prusa code
+ */
+
 #pragma once
 
+#include "../../lib/Marlin/Marlin/src/inc/MarlinConfig.h"
+#include "marlin_events.h"
+#include "marlin_vars.hpp"
+#include "marlin_errors.h"
+#include "client_fsm_types.h"
+
+#include <stddef.h>
 #include <optional>
 #include <atomic>
 
-#include "marlin_server.h"
+#include "marlin_server.hpp"
 #include "client_response.hpp"
 #include "fsm_types.hpp"
 #include "fsm_progress_type.hpp"
+// server flags
+// FIXME define the same type for these and marlin_server.flags
+static constexpr uint16_t MARLIN_SFLG_STARTED = 0x0001; // server started (set in marlin_server_init)
+static constexpr uint16_t MARLIN_SFLG_PROCESS = 0x0002; // loop processing in main thread is enabled
+static constexpr uint16_t MARLIN_SFLG_BUSY = 0x0004;    // loop is busy
+static constexpr uint16_t MARLIN_SFLG_PENDREQ = 0x0008; // pending request
+static constexpr uint16_t MARLIN_SFLG_EXCMODE = 0x0010; // exclusive mode enabled (currently used for selftest/wizard)
+
+// server variable update interval [ms]
+static constexpr uint8_t MARLIN_UPDATE_PERIOD = 100;
+
+typedef void(marlin_server_idle_t)(void);
```

Comment:
The `void` in params may be omitted now?

Reply:
I wanted to use alias, but I have forgot about it
Thx